pkg/attestation/crafter/materials/oci_image.go

```diff
@@ -60,13 +62,72 @@ func (i *OCIImageCrafter) Craft(_ context.Context, imageRef string) (*api.Attest
 	repoName := remoteRef.Repository.String()
 	i.logger.Debug().Str("name", repoName).Str("digest", remoteRef.DigestStr()).Msg("container image resolved")
 
+	// Check if the signature tag exists for the image digest
+	signDigest := i.CheckForSignature(ref, descriptor)
+
+	containerImage := &api.Attestation_Material_ContainerImage{
+		Id: i.input.Name, Name: repoName, Digest: remoteRef.DigestStr(), IsSubject: i.input.Output,
+		Tag: ref.Identifier(),
+	}
+
+	// If the signature digest exists, add it to the material
+	if signDigest != "" {
+		containerImage.SignatureDigest = signDigest
+	}
+
 	return &api.Attestation_Material{
 		MaterialType: i.input.Type,
-		M: &api.Attestation_Material_ContainerImage_{
-			ContainerImage: &api.Attestation_Material_ContainerImage{
-				Id: i.input.Name, Name: repoName, Digest: remoteRef.DigestStr(), IsSubject: i.input.Output,
-				Tag: ref.Identifier(),
-			},
-		},
+		M:            &api.Attestation_Material_ContainerImage_{ContainerImage: containerImage},
 	}, nil
 }
+
+// CheckForSignature checks if the signature tag exists for the image digest.
+// It first checks for a tag with the signature digest and, if not found, uses the referrers API to check for related material.
+func (i *OCIImageCrafter) CheckForSignature(originalRef name.Reference, originalDesc *remote.Descriptor) string {
+	trimmedDigest := strings.TrimPrefix(originalDesc.Digest.String(), "sha256:")
+	newTag := fmt.Sprintf("sha256-%s.sig", trimmedDigest)
+
+	// Create new reference for the signature tag
+	newRef, err := name.NewTag(fmt.Sprintf("%s:%s", originalRef.Context().Name(), newTag))
+	if err != nil {
+		i.logger.Error().Err(err).Msg("failed to create new tag reference")
+		return ""
+	}
+
+	// Try fetching the signature tag
+	if desc, err := remote.Head(newRef); err == nil {
```

Comment:
how is this performing authentication if credentials are passed?

Reply:
It's missing the keychain, I'll add it thanks!